src/main/java/gregtech/api/cover/CoverBehavior.java

```diff
@@ -209,6 +214,20 @@ public boolean isRemote() {
         return coverHolder.getWorld().isRemote;
     }
 
+    public Situation getSituation() {
+        return this.situation;
+    }
+
+    public void setSituation(Situation situation) {
+        this.situation = situation;
+    }
+
+    public void update(){
+        if (!isWorkingAllowed) {
+            setSituation(DISABLED_BY_CONTROLLER);
+        }
+    }
```

Comment:
I am not sure about this. All covers have to handle `isWorkingAllowed` on their own so there is probably no need for this.